workshops/web_chat/README.md

```diff
@@ -0,0 +1,782 @@
+
+- We'll need two rectangles, one large and one small - The large one will contain the messages sent and received (the message log) and the smaller one will contain controls and a space to type your message.
+
+![Large rectangle on top smaller rectangle below](https://cloud-91xu3gqm8.vercel.app/2weblayout2.png)
+
+- Inside our smaller rectangle, we'll need 3 things
+  - A field to type the user's name
+  - A field to type their message
+  - A button so they can send their message
+
+![Addition to previous image, three elements inside smaller rectangle](https://cloud-91xu3gqm8.vercel.app/3weblayout3.png)
+
+- We won't add stuff to the larger rectangle right now -- that'll happen when someone sends a message. Here's how a message will look
+
+![Message layout example inside larger rectangle, name on the left of the message rectangle and message content in the center](https://cloud-91xu3gqm8.vercel.app/4weblayout4.png)
+
+### Adding the HTML
+
+The HTML is the structure of the webpage. We won't be going into too much detail in this section -- to learn more about HTML and CSS, check out [this workshop on making your first website](https://workshops.hackclub.com/personal_website/). Add the following code to your HTML file
+
+```html
+<!DOCTYPE html>
+<html>
+<head></head>
+<body>
+    <div id="messageLog"></div>
+    <div id="controls">
+        <input id="name" placeholder="Your name">
+        <textarea id="message" placeholder="Type your message here"></textarea>
+        <button id="send">
+          Send
+        </button>
+    </div>
+</body>
+</html>
+```
+
+I'll quickly explain this code:
+
+- The biggest rectangle that contains all other rectangles we spoke about is the `<body>` tag
+  - The message log (Large rectangle present on top) is `<div id="messageLog"></div>` 
+  - The controls (Small rectangle below) is `<div id="controls">`
+    - The `<input>` is for the user to input their name
+    - The `<textarea>` (a multi-line `<input>`) is for the user to type their message
+    - The `<button>` will allow the user to send their message
+
+### Adding the CSS
+
+Since our frontend doesn't contain a lot of code, we'll be using an inline stylesheet - This means that all our CSS will be stored within the index.html file. For larger projects, you'll want to use [external stylesheets](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link). To write inline CSS, we'll create a `<style></style>` inside the `<head></head>`
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+  <style>
+    <!-- ADD YOUR INLINE STYLES HERE-->
+  </style>
+</head>
+<body>
+    <div id="messageLog"></div>
+    <div id="controls">
+        <input id="name" placeholder="Your name">
+        <textarea id="message" placeholder="Type your message here"></textarea>
+        <button id="send">
+          Send
+        </button>
+    </div>
+</body>
+</html>
+```
+
+We'll put all of our CSS here.
+
+- Add the following CSS into `<style></style>` inside your HTML file
```

Comment:
Can you rewrite this part of the workshop to include linking a CSS file and then put all of this code in a CSS file? I think using <style> tags in HTML is bad practice, and I don't want to potentially promote it as "the way to add styling" to beginners. You don't have to deeply explain how linking CSS files works, since that's already covered in the personal website workshop.